### ISSUE TITLE
Entrypoint should be renamed

### ISSUE THREAD

**Issue by waylan** [wilhelmer/mkdocs-localsearch#1]
I think its great that this plugin exists. However, I am concerned that it uses the same name as the builtin plugin: `search`. Note that the way entrypoints work, multiple entries can contain the same name, and all of them are returned as an iterable by `pkg_resources`. However, there is no guarantee in what order they will be listed (see documentation for [iter_entry_points][1] which states: *"Note that within the entry points advertised by an individual distribution, there is no particular ordering."*). Generally, the items are listed with the most recently installed listed first, but that isn't guaranteed either. Currently, MkDocs uses the first entry in the list. The fact that that results in your users getting your plugin is coincidental and rather fragile. 

There are a few possible problems with that approach:

1. If a third plugin used the same name, users will get a different plugin depending on which order they install them.
2. I haven't confirmed this, but I believe an upgrade of MkDocs may reorder the list, resulting in the builtin plugin being loaded.
3. MkDocs makes no promises that the current method of using the first installed plugin will work. A future version could raise an error if it detects more than one plugin with the same name assigned to the `mkdocs.plugins` group. 
4. An update to `pkg_resources` could result in the entries being returned in a different order.
5. The same project installed on different systems could get the entries in a different order.

My suggestion would be to rename the entrypoint to something unique. Perhaps `localsearch`? And yes, that means your users would all need to include the plugin in their `mkdocs.yml` config file.

```yml
plugins:
    - localsearch
```

[1]: https://setuptools.readthedocs.io/en/latest/pkg_resources.html#basic-workingset-methods

**Comment by wilhelmer** [comment]
Thanks for the feedback. I actually named it `localsearch` in the first place, but that didn't work. The plugin was ignored and the builtin search was used instead.

Then I thought "hmm, maybe the way third-party search plugins are supposed to work is by naming them exactly like the builtin plugin", did that and boom, it worked. Because the plugin is kinda "hacky" anyway, I left it like that although it didn't feel right.

I'll retry and look into that. Maybe it was an issue with the Material theme I use.

**Comment by waylan** [comment]
The Material theme does its own thing with search (it provides is own JS rather than using the plugin provided JS). I would test with one of the builtin themes first. Once you get that working, then move to on to getting it to work with Material. However, I wouldn't be surprised if the Material theme would need some changes to also support your plugin.

**Comment by wilhelmer** [comment]
The builtin `MkDocs` theme doesn't use `fetch()`, so the shim doesn't work. I could extend the shim or modify `worker.js`, but I don't have time for this. 

However, I got the plugin to work with Material and a renamed entrypoint. There's just a small change needed in the `header.html`.

I have published v0.3.0 with the following changes:

- Added note to `README.md` that the plugin currently only works with the Material theme.
- Renamed entrypoint to `localsearch` as suggested.
- Updated installation instructions. 

Might not be the best (and certainly not a universal) solution, but I assume the issue can be closed?
